File: app/src/main/java/com/udacity/sandwichclub/utils/JsonUtils.java
package com.udacity.sandwichclub.utils;

import com.udacity.sandwichclub.model.Sandwich;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class JsonUtils {

    public static Sandwich parseSandwichJson(String json) {

        Sandwich sandwich = null;

        try {
            //create JSON Sandwich object to get JSON data
            JSONObject sandwichObject = new JSONObject(json);

            //get JSON objects from JSON data
            JSONObject sandwichName = sandwichObject.optJSONObject("name");

            //get JSON string from JSON data
             String sandwichMainName = sandwichName.optString("mainName");

            //get JSON Array and convert JSON array to java list
            JSONArray alsoknownasArray = sandwichName.optJSONArray("alsoKnownAs");
            List<String> sandwichAlsoKnownAs = convertArray(alsoknownasArray);


            //get rest of JSON strings from JSON data
            String sandwichPlaceOfOrigin = sandwichObject.optString("placeOfOrigin");
            String sandwichDescription = sandwichObject.optString("description");
            String sandwichImage = sandwichObject.optString("image");

            //get JSON Array and convert JSON array to java list
            JSONArray ingredientsArray = sandwichObject.optJSONArray("ingredients");
            List<String> sandwichIngredients = convertArray(ingredientsArray);

            //create a new sandwhich object to return JSON data
            sandwich = new Sandwich(sandwichMainName, sandwichAlsoKnownAs, sandwichPlaceOfOrigin, sandwichDescription, sandwichImage, sandwichIngredients);


        } catch (JSONException ex) {
            ex.printStackTrace();
        }

        //return newly created sandwhich object
        return sandwich;

    }
        //create helper method for convert arrays
        private static List<String> convertArray(JSONArray jsonArray) throws JSONException {
            int arrayLength = jsonArray.length();
            List<String> convertedArray = new ArrayList<>();

            // iterate through array and  add to list
            for (int i = 0; i < arrayLength; i++) {
                convertedArray.add(jsonArray.getString(i));

            }

            return convertedArray;
        }

    }

File: app/src/main/java/com/udacity/sandwichclub/utils/JsonUtils.kt
package com.udacity.sandwichclub.utils

import com.udacity.sandwichclub.model.Sandwich

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

import java.util.ArrayList

object JsonUtils {

    fun parseSandwichJson(json: String): Sandwich? {

        var sandwich: Sandwich? = null

        try {
            //create JSON Sandwich object to get JSON data
            val sandwichObject = JSONObject(json)

            //get JSON objects from JSON data
            val sandwichName = sandwichObject.optJSONObject("name")

            //get JSON string from JSON data
            val sandwichMainName = sandwichName.optString("mainName")

            //get JSON Array and convert JSON array to java list
            val alsoknownasArray = sandwichName.optJSONArray("alsoKnownAs")
            val sandwichAlsoKnownAs = convertArray(alsoknownasArray)


            //get rest of JSON strings from JSON data
            val sandwichPlaceOfOrigin = sandwichObject.optString("placeOfOrigin")
            val sandwichDescription = sandwichObject.optString("description")
            val sandwichImage = sandwichObject.optString("image")

            //get JSON Array and convert JSON array to java list
            val ingredientsArray = sandwichObject.optJSONArray("ingredients")
            val sandwichIngredients = convertArray(ingredientsArray)

            //create a new sandwhich object to return JSON data
            sandwich = Sandwich(sandwichMainName, sandwichAlsoKnownAs, sandwichPlaceOfOrigin, sandwichDescription, sandwichImage, sandwichIngredients)


        } catch (ex: JSONException) {
            ex.printStackTrace()
        }

        //return newly created sandwich object
        return sandwich

    }

    //create helper method for convert arrays
    @Throws(JSONException::class)
    private fun convertArray(jsonArray: JSONArray): List<String> {
        val arrayLength = jsonArray.length()
        val convertedArray = ArrayList<String>()

        // iterate through array and  add to list
        for (i in 0 until arrayLength) {
            convertedArray.add(jsonArray.getString(i))

        }

        return convertedArray
    }

}
